src/rust/engine/process_execution/src/remote.rs

```diff
@@ -1292,6 +1401,33 @@ pub fn extract_output_files(
   .to_boxed()
 }
 
+/// Apply REAPI request metadata header to a `tonic::Request`.
+fn apply_headers<T>(mut request: Request<T>, build_id: &str) -> Request<T> {
+  let reapi_request_metadata = remexec::RequestMetadata {
+    tool_details: Some(remexec::ToolDetails {
+      tool_name: "pants".into(),
+      ..remexec::ToolDetails::default()
+    }),
+    tool_invocation_id: build_id.to_string(),
+    ..remexec::RequestMetadata::default()
+  };
+
+  let bytes = {
+    let mut buf = BytesMut::with_capacity(reapi_request_metadata.encoded_len());
+    // TODO(tonic): Handle encode errors?
```

Comment:
Yeah, I assume all callers would be ~trivial to switch over to take a `Result`

Reply:
Will do in a separate PR to make review easier.